aws-http-auth/sigv4a/e2e_test.go

```diff
@@ -0,0 +1,535 @@
+	method := http.MethodGet
+	path := "/v20180820/async-requests/mrap/" + in.RequestToken
+	if err := c.do(ctx, c.AccountID, method, path, in, &out); err != nil {
+		return nil, err
+	}
+	return &out, nil
+}
+
+type DescribeMRAPOperationInput struct {
+	RequestToken string
+}
+
+func (i *DescribeMRAPOperationInput) ToXML() []byte {
+	return []byte("")
+}
+
+type DescribeMRAPOperationOutput struct {
+	AsyncOperation struct {
+		RequestStatus string
+	}
+}
+
+func (c *S3ControlClient) DeleteMRAP(ctx context.Context, in *DeleteMRAPInput) (*DeleteMRAPOutput, error) {
+	var out DeleteMRAPOutput
+	method := http.MethodPost
+	path := "/v20180820/async-requests/mrap/delete"
+	if err := c.do(ctx, c.AccountID, method, path, in, &out); err != nil {
+		return nil, err
+	}
+	return &out, nil
+}
+
+type DeleteMRAPInput struct {
+	Name string
+}
+
+func (i *DeleteMRAPInput) ToXML() []byte {
+	const tmpl = `
+<DeleteMultiRegionAccessPointRequest xmlns="http://awss3control.amazonaws.com/doc/2018-08-20/">
+	<ClientToken>%s</ClientToken>
+	<Details>
+		<Name>%s</Name>
+	</Details>
+</DeleteMultiRegionAccessPointRequest>`
+
+	token := fmt.Sprintf("%d", rand.Int31())
+	return []byte(fmt.Sprintf(tmpl, token, i.Name))
+}
+
+type DeleteMRAPOutput struct {
+	RequestToken string `xml:"RequestTokenARN"`
+}
+
+func (c *S3ControlClient) do(ctx context.Context, accountID, method, path string, in ToXML, out any) error {
+	// init
+	endpoint := fmt.Sprintf("https://%s.s3-control.us-west-2.amazonaws.com", accountID)
+	req, err := http.NewRequestWithContext(ctx, method, endpoint, http.NoBody)
+	if err != nil {
+		return fmt.Errorf("new http request: %w", err)
+	}
+
+	// serialize
+	req.URL.Path = path
+	req.Header.Set("Content-Type", "application/xml")
+	req.Header.Set("X-Amz-Account-Id", accountID)
+	payload := in.ToXML()
+	req.Body = closer{bytes.NewReader(payload)}
+	req.ContentLength = int64(len(payload))
+
+	// sign
+	err = c.Signer.SignRequest(&sigv4.SignRequestInput{
+		Request:     req,
+		Credentials: c.Credentials,
+		Service:     "s3",
+		Region:      "us-west-2",
+	})
+	if err != nil {
+		return fmt.Errorf("sign request: %w", err)
+	}
+
+	// round-trip
+	resp, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("do request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	// deserialize
+	data, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("read response body: %w", err)
+	}
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		return fmt.Errorf("request error: %s: %s", resp.Status, data)
+	}
+	if len(data) == 0 {
+		return nil
+	}
+	if err := xml.Unmarshal(data, out); err != nil {
+		return fmt.Errorf("deserialize response: %w", err)
+	}
+
+	return nil
+}
+
+// WARNING: this test takes a while, because creating an MRAP is asynchronous
+// and slow
+//
+// 1. creates a bucket in us-east-1
+// 2. creates an MRAP that points to that bucket
+// 3. polls MRAP status until created
+// 4. puts object to MRAP
```

Comment:
So I suspect is only up to this point that we check sigv4a, right?

Depending on how much "a while" is "a while", I could lean to instead create a bucket if not exists without deleting the bucket at the end, so we skip the lag of MRAP creation. This still validates what we want to validate, right?

Reply:
2-3 minutes in my own testing, it seems.

Reply:
Yeah I did consider adding an env for basically "MRAP_ALIAS" - i.e. if that's set, just bypass the whole setup/teardown and just call against that - but by the time I got this test written I was so done with it that I moved on to other stuff :joy:

We can put that in now if you feel strongly that it should be an option.